### PR TITLE
feat(cloudserver): recover panic

### PR DESCRIPTION
**Pull request by dojusa**

### CODE REVIEW

File: cloudserver/http.go [hunk 1]
```diff
@@ -2,16 +2,27 @@ package cloudserver
...
 // HTTPServer provides HTTP server middleware.
 func (i *Middleware) HTTPServer(next http.Handler) http.Handler {
-	return i.httpTimeout(next)
-}
-
-func (i *Middleware) httpTimeout(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		defer func() {
+			if r := recover(); r != nil {
+				writer.WriteHeader(http.StatusInternalServerError)
```
Comment (odsod) [inline cloudserver/http.go:17]: if we have already written a header, will this be a no-op or an error?

Comment (dojusa) [reply cloudserver/http.go:17]: @odsod  it will be a no-op. Added some unit tests to verify it.

**Review by odsod — APPROVED** [review]
@dojusa Nice one!

File: cloudserver/http.go (hunk 1 above)
Comment (odsod) [reply cloudserver/http.go:17]: Brilliant! :rocket:

**Pull request merged**